ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -470,6 +464,105 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
     }
   }
 
+  "authorizeSubmitters" should {
+    val alice = "alice"
+    val bob = "bob"
+    val emma = "emma"
+    val participantId = 0
+    val otherParticipantId = 1
+    def partyAllocation(party: String, participantId: Int): DamlPartyAllocation =
+      DamlPartyAllocation
+        .newBuilder()
+        .setParticipantId(mkParticipantId(participantId))
+        .setDisplayName(party)
+        .build()
+
+    def hostedParty(party: String): DamlPartyAllocation = partyAllocation(party, participantId)
+    def notHostedParty(party: String): DamlPartyAllocation =
+      partyAllocation(party, otherParticipantId)
+    def createInputs(
+        inputs: (String, Option[DamlPartyAllocation])*): Map[DamlStateKey, Option[DamlStateValue]] =
+      inputs.map {
+        case (party, partyAllocation) =>
+          DamlStateKey.newBuilder().setParty(party).build() -> partyAllocation.map(
+            DamlStateValue.newBuilder().setParty(_).build())
+      }.toMap
```

Comment:
Please move helpers to the companion object, or if that's not possible, to the bottom of the class or "should" block.

Reply:
I've moved those to the bottom of `should` block.